### PR TITLE
Edit pass, `xref`, and NuGet links, etc.

### LINKED ISSUE #32499: Update the "What's new in Orleans" article
Update the "What's new in Orleans" article based on my experiences from upgrading all of the dotnet/orleans/samples bits in https://github.com/dotnet/orleans/issues/8035.

- `KnownAssembly` should be `GenerateCodeForDeclaringAssembly` instead.

https://learn.microsoft.com/dotnet/orleans/whats-new-in-orleans

Also, include a request from #32561 to ensure that the article is easier to follow.

---
[Associated WorkItem - 34667](https://dev.azure.com/msft-skilling/Content/_workitems/edit/34667)

### LINKED ISSUE #32561: Please include namespaces
Please include the namespaces / package names in code examples to save us going on a wild goose chase. 

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 4aab851b-48df-ff93-da26-dfb9230c07bd
* Version Independent ID: cfd3e88c-ff32-1250-bd69-f82ca5571725
* Content: [What's new in Orleans](https://learn.microsoft.com/en-us/dotnet/orleans/whats-new-in-orleans)
* Content Source: [docs/orleans/whats-new-in-orleans.md](https://github.com/dotnet/docs/blob/main/docs/orleans/whats-new-in-orleans.md)
* Product: **dotnet-orleans**
* GitHub Login: @IEvangelist
* Microsoft Alias: **dapine**

---
[Associated WorkItem - 37090](https://dev.azure.com/msft-skilling/Content/_workitems/edit/37090)

### PR DESCRIPTION
**Pull request by IEvangelist**
## Summary

- Edit pass
- Use `xref` where appropriate to link to Orleans API docs
- Add NuGet links and namespaces where applicable
- Grammatical error corrections
- Fix heading levels

Fixes #32499 and fixes #32561

## Internal preview

- [What's new in Orleans](https://review.learn.microsoft.com/en-us/dotnet/orleans/whats-new-in-orleans?branch=pr-en-us-32582)

### CODE REVIEW

File: docs/orleans/whats-new-in-orleans.md [hunk 1]
```diff
@@ -467,21 +504,21 @@ builder.AddMemoryStreams<DefaultMemoryMessageBodySerializer>(
     });
 ```
 
-### Open Telemetry
+## Open Telemetry
 
 The telemetry system has been updated in Orleans 7.0 and the previous system has been removed in favor of standardized .NET APIs such as .NET Metrics for metrics and ActivitySource for tracing.
 
 As a part of this, the existing `Microsoft.Orleans.TelemetryConsumers.*` packages have been removed. We are considering introducing a new set of packages to streamline the process of integrating the metrics emitted by Orleans into your monitoring solution of choice. As always, feedback and contributions are welcome.
```
Comment (ReubenBond) [inline docs/orleans/whats-new-in-orleans.md:511]: We should include a bit here about the replacement. For counters, the dotnet-counters tool can be used to monitor them: `dotnet counters monitor -n MyApp --counters Microsoft.Orleans`. Similarly, OpenTelemetry metrics can add the `Microsoft.Orleans` meters. For distributed tracing, OpenTelemetry can be used to monitor `Microsoft.Orleans.Runtime` and `Microsoft.Orleans.Application` as in the .NET Conf video: https://youtu.be/WUZw78P-Zg8?t=1382 and in the corresponding [sample code here](https://github.com/ReubenBond/orleans/blob/ad86ca49688769e7c2f64a748f60611691f889b6/samples/GPSTracker/GPSTracker.Service/Program.cs#L14):

```csharp
builder.Services.AddOpenTelemetryMetrics(metrics =>
{
    metrics
        .AddPrometheusExporter()
        .AddMeter("Microsoft.Orleans");

    // Uncomment this to export metrics to Azure Monitor
    //metrics.AddAzureMonitorMetricExporter(config => config.ConnectionString = Environment.GetEnvironmentVariable("AZMONCS"))
});
```

Distributed Tracing can be enabled by configuring Open Telemetry:
```csharp
builder.Services.AddOpenTelemetryTracing(tracing =>
{
    // Set a service name
    tracing.SetResourceBuilder(ResourceBuilder.CreateDefault().AddService(serviceName: "GPSTracker", serviceVersion: "1.0"));

    tracing.AddAspNetCoreInstrumentation();
    tracing.AddSource("Microsoft.Orleans.Runtime");
    tracing.AddSource("Microsoft.Orleans.Application");

    tracing.AddZipkinExporter(zipkin =>
    {
        zipkin.Endpoint = new Uri("http://localhost:9411/api/v2/spans");
    });
});
```
and the silo builder:
```csharp

builder.Host.UseOrleans((ctx, siloBuilder) =>
{
    // ... snip ...
    siloBuilder.AddActivityPropagation();
});
```

**Review by ReubenBond — APPROVED** [review]
I left some comments, too

File: docs/orleans/whats-new-in-orleans.md (hunk 1 above)
Comment (IEvangelist) [reply docs/orleans/whats-new-in-orleans.md:511]: Added in [59aae8f](https://github.com/dotnet/docs/pull/32582/commits/59aae8f25cc1fcb757e0f23eaa0305179e289e0a)

**Review by ReubenBond — APPROVED** [review]

**Review by CamSoper — APPROVED** [review]
LGTM @IEvangelist

**Pull request merged**